### PR TITLE
[solution]: zh/medium-anyof

### PR DESCRIPTION
**Pull request by charleserious**
# Challenge [zh/medium-anyof]

## Checklist

- [x] Make sure that you are opening the PR with a single solution (one solution per PR)
- [x] The solution must have at least several references for the readers to dive in depth

### CODE REVIEW

File: zh/medium-anyof.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+我们从推断元组中的单一元素开始，并推断族中其余的元素：
+
+```typescript
+type AnyOf<T extends readonly any[]> = T extends [infer H, ...infer T]
+  ? never
+  : never;
+```
+
+我们如何检查推断的元素 `H` 是否为 false-y？
+首先，我们可以构造一个表示 falsy-y 的类型。
```
Comment (likui628) [inline zh/medium-anyof.md:57]: typo: falsy-y

File: zh/medium-anyof.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+```typescript
+type AnyOf<T extends readonly any[]> = T extends [infer H, ...infer T]
+  ? H extends Falsy
+    ? AnyOf<T>
+    : never
+  : never;
+```
+
+最后，一量我们看到有元素不是 false-y，就意味着它是 true-y。
```
Comment (likui628) [inline zh/medium-anyof.md:86]: 一量 -> 一旦

File: zh/medium-anyof.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+由于我们已经知道存在 true-y 元素，因此继续递归就没有意思了。
+所以我们只需要通过返回 `true` 类型字面量来既定出递归：
```
Comment (likui628) [inline zh/medium-anyof.md:88]: 既定出递归 -> 退出递归 or 跳出递归

File: zh/medium-anyof.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+```typescript
+type AnyOf<T extends readonly any[]> = T extends [infer H, ...infer T]
+  ? H extends Falsy
+    ? AnyOf<T>
+    : true
+  : never;
+```
+
+最后的状态是当我们有一个空无组时。
```
Comment (likui628) [inline zh/medium-anyof.md:98]: 空无组->空元组

**Review by likui628 — DISMISSED** [review]

File: zh/medium-anyof.md (hunk 1 above)
Comment (charleserious) [reply zh/medium-anyof.md:57]: resolved, thx.

File: zh/medium-anyof.md (hunk 2 above)
Comment (charleserious) [reply zh/medium-anyof.md:86]: Sorry for that.

File: zh/medium-anyof.md (hunk 3 above)
Comment (charleserious) [reply zh/medium-anyof.md:88]: Sorry for the careless....

File: zh/medium-anyof.md (hunk 4 above)
Comment (charleserious) [reply zh/medium-anyof.md:98]: typo....

**Review by likui628 — APPROVED** [review]

**Pull request merged**